curated_transformers/layers/attention.py

```diff
@@ -877,34 +909,22 @@ def forward(
             causal_mask = create_causal_mask(query, key)
             combined_mask = combined_mask.merge_mask(causal_mask)
 
-        if _TORCH_SDP.get():
-            attn_mask = combined_mask.logit_mask(query.dtype)
-
-            # Add AliBi to the logit mask
-            if self.use_alibi:
-                assert self.attention.linear_biases is not None
-                biases = self.attention.linear_biases.calculate_biases(key.size(-2)).to(
-                    dtype=query.dtype, device=query.device
-                )
-                bool_mask = combined_mask.bool_mask
-                attn_mask = torch.where(bool_mask, biases, attn_mask)
-
-            # We can't pass a bool mask, because it is currently broken:
-            # https://github.com/pytorch/pytorch/issues/103749
-            attn = F.scaled_dot_product_attention(
-                query=query,
-                key=key,
-                value=value,
-                attn_mask=attn_mask,
-                dropout_p=self.dropout_prob if self.training else 0.0,
-            )
-        else:
-            attn = self.attention(
-                query=query,
-                key=key,
-                value=value,
-                attention_mask=combined_mask,
+        if _TORCH_SDP.get() and not isinstance(
+            self.attention_scorer, ScaledDotProductAttention
+        ):
+            warn_msg = (
```

Comment:
Not sure if we want this warning. People could be running multiple models and have them all scoped in the context manager to enable Torch SDP for all supported models.

Also, someone could make another scorer class that also uses Torch SDP.

Reply:
Good point. I'll remove it.